agent/consul/fsm/snapshot_oss.go

```diff
@@ -309,6 +310,10 @@ func (s *snapshot) persistConnectCAConfig(sink raft.SnapshotSink,
 	if err != nil {
 		return err
 	}
+	// Make sure we don't write a nil config out to a snapshot.
+	if config == nil {
+		return nil
+	}
```

Comment:
I really think these changes need tests... 

I think something like `TestFSM_BadRestore_OSS` but for the happy path but with no CA config to prove that a nil config never makes it into the Snapshot.